test/common/ops/hourglassPurchase.js

```diff
@@ -0,0 +1,136 @@
+import hourglassPurchase from '../../../common/script/ops/hourglassPurchase';
+import {
+  BadRequest,
+  NotAuthorized,
+} from '../../../common/script/libs/errors';
+import i18n from '../../../common/script/i18n';
+import content from '../../../common/script/content/index';
+import {
+  generateUser,
+} from '../../helpers/common.helper';
+
+describe('user.ops.hourglassPurchase', () => {
+  let user;
+
+  beforeEach(() => {
+    user = generateUser();
+  });
+
+  context('failure conditions', () => {
+    it('return error when key is not provided', () => {
+      try {
```

Comment:
yeah I was wondering how to avoid skipping tests if an error wasn't thrown, using `done` is a good idea! Regarding `expect.throw` it works but IMO it's very confusing